_posts/2020-12-01-configuring-and-suppressing-warnings.md

```diff
@@ -0,0 +1,192 @@
+
+Multiple filters can be combined using `&` to narrow down the selection. For example, the following configuration turns deprecation warnings for `scala.Predef` into errors:
+
+```scala
+$> scalac '-Wconf:cat=deprecation&origin=scala\.Predef\..*:e' Test.scala
+Test.scala:4: error: method → in class ArrowAssoc is deprecated [...]
+  def f() = 1 → 2
+              ^
+1 error
+```
+
+Note that the `-Wconf:...` compiler argument is between quotes (`'`) in the command line, which prevents the shell from interpreting characters like `&` or `*`.
+
+For some of the filters the syntax is not trivial, so we look at them in more detail.
+
+* **Message category**: Every message has a category that is displayed in verbose mode (`-Wconf:any:wv`). The `-Wconf:help` option displays the full list of available categories. For example, every `-Xlint` warning has its own category (`lint-infer-any`), the super-category `lint` matches all lint warnings.
+* **Source file**: By default, the source file filter is a regex that must match the file path relative to any path segment. For example, `b/.*Test.scala` matches `/a/b/XTest.scala` but not `/ab/Test.scala`. If the `-rootdir` compiler option is specified, the regex must match the file path relative to that root directory.
+* **Since version for deprecations**: In a `since<1.24` filter expression, valid operators are `<`, `=` and `>` and valid version numbers are `N`, `N.M` and `N.M.P`. Because the `since` annotation argument can contain arbitrary text, the first version number found in the text is used for the comparison, for example `1.2.3` in `@deprecated("", "some lib 1.2.3-foo")`.
+
+## Local warning suppression using `@nowarn`
+
+The [`@nowarn` annotation](https://www.scala-lang.org/api/current/scala/annotation/nowarn.html) allows suppressing warnings locally within a source file. It can be applied to method or class definitions, or to individual expressions using the ascription syntax `expression: @nowarn`.
+
+```scala
+scala> @deprecated def dpr = 0
+def dpr: Int
+
+// don't issue any warnings for code in method `f`
+scala> @annotation.nowarn def f = { 1; dpr }
+def f: Int
+
+// don't issue the "a pure expression does nothing in statement position" warning
+scala> def f = {
+     |   1: @annotation.nowarn 
+     |   dpr
+     | }
+         ^
+       warning: method dpr is deprecated
+def f: Int
+```
+
+
+The `@nowarn` annotation has an optional value parameter to silence warnings selectively, where the syntax is the same as a filter expression of the `-Wconf` compiler option.
+
+```scala
+scala> @annotation.nowarn("msg=pure expression does nothing") def f = { 1; dpr }
+                                                                           ^
+       warning: method dpr is deprecated
+def f: Int
+```
+
+To ensure that `@nowarn` annotations actually suppress warnings, enable `-Xlint:unused` or `-Wunused:nowarn`.
```

Comment:
I think you probably need to explain this a little more, I'm not confident it will be clear to all readers.